### DISCUSSION TITLE
oidc_cache_set: could NOT store 797 bytes in shm cache backend for key

### DISCUSSION THREAD

**Discussion by akasha2z** [OpenIDC/mod_auth_openidc#971] [Q&A]
Trying to configure the module for remote validation and use cache for tokens (OIDCOAuthTokenIntrospectionInterval param)
Seeing this error - oidc_cache_set: could NOT store 797 bytes in shm cache backend for key

I am still able to authenticate the users but tokens are not cached.

**Comment by zandbelt** [comment]
most probably you're using an (too) old version of mod_auth_openidc

**Reply by akasha2z** [reply]
Even after upgrading to latest version, I get this error
oidc_cache_shm_get_key: could not construct cache key since key size is too large (1409 >= 512)
I get that token size is large.

Is there a way to cache this anyway?

**Reply by zandbelt** [reply]
yes, set `OIDCCacheEncrypt On`

**Reply by akasha2z** [reply]
Doesn't seem to be working with apache. Apache restarts when this parameter is set.

**Reply by zandbelt** [reply]
which platform are you on, where did you get the binaries from and what does the server log say after setting `LogLevel auth_openidc:debug` ?

**Reply by akasha2z** [reply]
On linux, FreeBSD. I had the code compiled on it from source. Remote and local validation works. For local validation using JWKS uri, I can see the kids are cached successfully.
Now trying to cache the access tokens for remote introspection method.
I see this in the log  and then Apache restarts after this -
util.c(1077): [client 10.249.153.171:57114] oidc_util_http_call: response=<response from IDP, token validated successfully> 
common.c(325): [client 10.249.153.171:57114] oidc_cache_set: enter: https://<IDP ip and realm>/.well-known/openid-configuration (section=o, len=6075, encrypt=1, ttl(s)=86399, type=shm)

**Reply by zandbelt** [reply]
I cannot reproduce this on any platform locally and I don't have access to FreeBSD so I guess you're on your own; most probably an OpenSSL issue

**Reply by akasha2z** [reply]
Thanks for your input

**Reply by akasha2z** [reply]
Crash happens in oidc_jwt_sign(), when calling the cjose module's method cjose_jws_sign()

**Comment by akasha2z** [comment]
Ya it's old version - 2.4.3
Will this be solved with the latest version?
